spectator-api/src/main/java/com/netflix/spectator/api/ActivityCounter.java

```diff
@@ -0,0 +1,65 @@
+package com.netflix.spectator.api;
+
+import java.util.Collections;
+import java.util.concurrent.atomic.AtomicLong;
+
+/**
+ * A counter that also keeps track of the time since last update.
+ */
+public final class ActivityCounter implements Counter {
+
+  /** Create a new instance. */
+  public static ActivityCounter get(Registry registry, Id id) {
+    return new ActivityCounter(registry, id);
+  }
+
+  private static final Tag INTERVAL_TAG = new BasicTag("statistic", "interval");
+
+  private final Registry registry;
+  private final Id id;
+  private final Counter counter;
+  private final AtomicLong lastUpdated;
+
+  /**
+   * Create a new ActivityCounter using the given registry and base id.
+   */
+  ActivityCounter(Registry registry, Id id) {
+    this.registry = registry;
+    this.id = id;
+    this.counter = registry.counter(id.withTag(Statistic.count));
+    this.lastUpdated = new AtomicLong(0L);
+    registry.gauge(id.withTag(INTERVAL_TAG), lastUpdated, Functions.age(registry.clock()));
```

Comment:
I'm not sure this will work as the user would expect. If they do not keep a strong reference to the activity counter then the gauge will go away. I could see users expecting to do:

```java
Id id = ...
ActivityCounter.get(registry, id).increment();
```